### PR TITLE
Fix problems found on buildbots

### PR DESCRIPTION
**Pull request by abergeron**

### CODE REVIEW

**Comment by abergeron** [comment]
jenkins test win

**Comment by abergeron** [comment]
The last failing test is just that we are unlucky with the seed and the random indices that were drawn happen to be sorted.

**Comment by abergeron** [comment]
jenkins test windows

**Comment by abergeron** [comment]
I fixed the artifacts and upload paths on CircleCI as well.

**Comment by abergeron** [comment]
jenkins test windows

**Comment by abergeron** [comment]
jenkins test windows (the test server decided to reboot for updates)

**Comment by nouiz** [comment]
still many failures.

File: theano/tensor/extra_ops.py [hunk 1]
```diff
@@ -1218,7 +1218,7 @@ def perform(self, node, inp, out):
         res = np.unravel_index(indices, dims)
         assert len(res) == len(out)
         for i in xrange(len(out)):
-            out[i][0] = res[i]
+            out[i][0] = theano._asarray(res[i])
```
Comment (nouiz) [inline theano/tensor/extra_ops.py]: This request a dtype parameter. Idem bellow. I did that in a commit to this PR.

**Pull request merged**